types/react/index.d.ts

```diff
@@ -560,6 +560,121 @@ declare namespace React {
 
     function forwardRef<T, P = {}>(Component: RefForwardingComponent<T, P>): ComponentType<P & ClassAttributes<T>>;
 
+    //
+    // React Hooks
+    // ----------------------------------------------------------------------
+
+    // based on the code in https://github.com/facebook/react/pull/13968
+
+    // Unlike the class component setState, the updates are not allowed to be partial
+    type SetStateAction<S> = S | ((prevState: S) => S);
+    // this technically does accept a second argument, but it's already under a deprecation warning
+    // and it's not even released so probably better to not define it.
+    type Dispatch<A> = (value: A) => void;
+    // Unlike redux, the actions _can_ be anything
+    type Reducer<S, A> = (prevState: S, action: A) => S;
+    // The identity check is done with the SameValue algorithm (Object.is), which is stricter than ===
+    type IdentityCheckInputList = ReadonlyArray<any>;
+
+    // NOTE: the effect callbacks are actually allowed to return anything, but functions are treated
+    // specially. I don't think it's intended to accept async functions, should use throw instead of
+    // await to properly engage Suspense.
+    type EffectCallback = () => (void | (() => void));
+
+    interface MutableRefObject<T> {
+        current: T | null;
+    }
+
+    /**
+     * Accepts a context object (the value returned from `React.createContext`) and returns the current
+     * context value, as given by the nearest context provider for the given context.
+     *
+     * @see https://reactjs.org/docs/hooks-reference.html#usecontext
+     */
+    function useContext<T>(context: Context<T> | Consumer<T> /*, (not public API) observedBits?: number|boolean */): T;
```

Comment:
Since these are all proposed APIs, how would you feel about marking them as `@experimental` or `@alpha`?

Reply:
Please remove Consumer from there. Even though it works, for now, it shows a warning and should be used like that.